tianshou/data/batch.py

```diff
@@ -740,23 +902,18 @@ def shape(self) -> List[int]:
                     data_shape.append(list(obj.shape))
                 except AttributeError:
                     data_shape.append([])
-            return list(map(min, zip(*data_shape))) if len(data_shape) > 1 \
-                else data_shape[0]
+            return (
+                list(map(min, zip(*data_shape)))
+                if len(data_shape) > 1 else data_shape[0]
+            )
 
-    def split(self,
+    def split(self: TBatch,
```

Comment:
It's necessary for most methods that returns self, otherwise the annotation fails on child classes.

```python

class A:
    def return_self(self) -> 'A':
         return self

class B(A):
    pass
```

Without a `TypeA` generic type, you would incorrectly get that `B().return_self()` returns an instance of `A`.

See also a longer description [here](https://peps.python.org/pep-0673/), where the `Self` type was introduced to solve the rather ugly workaround with `TypeA`. Unfortunately, it's only in python 3.11, so unless we deprecate everything before that in tianshou, we can't use it